runtime/include/qio/qio_error.h

```diff
@@ -48,12 +48,29 @@ struct qio_err_s {
 // NULL means no error.
 typedef const struct qio_err_s* qioerr;
 
+struct qio_err_s* qio_error_get_base(void);
+
 // on linux, normal error codes are < 255
 enum { EXTEND_ERROR_OFFSET = 1000 };
 // on linux, these are -1 to -100.
 enum { GAI_ERROR_OFFSET = (EXTEND_ERROR_OFFSET+2000) };
 enum { MAX_ERROR_OFFSET = GAI_ERROR_OFFSET+1000 };
 
+static inline
```

Comment:
1) It would be good to put a nice big block comment before these two functions explaining what is going on, why the solution chosen is the right way to go, and what could cause it to break in the future (for example, if separate compilation for Chapel modules ever becomes a reality).
2) I assume all these are static inline in the .h file in order to improve performance. But is this really a performance critical area? The implementation borders on the diabolical, and it would be good to hide the details in a .c file instead of publicizing them in the interface .h file if at all possible.